src/tiledb/cloud/client.py

```diff
@@ -529,20 +538,23 @@ def find_organization_or_user_for_default_charges(user):
         (organization or user if not part of any organization)
     """
 
-    namespace_to_charge = user.username
-
-    if (
-        user.default_namespace_charged is not None
-        and user.default_namespace_charged != ""
-    ):
+    if user.default_namespace_charged:
         return user.default_namespace_charged
 
     for org in user.organizations:
-        if org.organization_name != "public":
-            namespace_to_charge = org.organization_name
-            break
-
-    return namespace_to_charge
+        if org.organization_name == "public":
+            continue
+        if required_action is None or required_action in org.allowed_actions:
+            return org.organization_name
+
+    if required_action is None or required_action in user.allowed_actions:
+        return user.username
+
+    raise Exception(
+        f"user {user.username} does not belong to "
+        + f"an organization supporting {required_action}. "
+        + "Please check the default_namespace in the console"
```

Comment:
Since adjacent strings are automatically joined together, the + signs are not needed here.

```suggestion
        f"an organization supporting {required_action}. "
        "Please check the default_namespace in the console"
```